incubator/hnc/internal/reconcilers/object.go

```diff
@@ -540,6 +540,25 @@ func (r *ObjectReconciler) writeObject(ctx context.Context, log logr.Logger, ins
 	if exist {
 		log.Info("Updating")
 		err = r.Update(ctx, inst)
+		// RoleBindings can't have their Roles changed after they're created. 
+		// If an RB was quickly delete and re-created in an ancestor namespace 
+		// - fast enough that by the time that HNC notices, the new RB exists - 
+		// HNC could see it as an update (not a delete + create) and attempt to 
+		// update the RBs in all descendant namespaces, and this will fail. 
+		// In order to handle this case, we try to delete and re-create the 
+		// rolebinding here
+		if err != nil && errors.IsInvalid(err) {
+			if err = r.Delete(ctx, inst); err == nil {
+				err = r.Create(ctx, inst)
+				if err != nil {
+					log.Info("Unable to create new object.")
+				} else {
+					log.Info("Successfully delete and create object.")
+				}
```

Comment:
We usually don't log _successful_ operations, but in this case we might want to say something like "Couldn't update object but delete/create worked" to indicate that something unusual happened.

Reply:
Done